pkg/resources/materialized_view_acceptance_test.go

```diff
@@ -0,0 +1,109 @@
+)
+
+func TestAcc_MaterializedView(t *testing.T) {
+	dbName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	schemaName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	tableName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	viewName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+
+	resource.ParallelTest(t, resource.TestCase{
+		Providers: providers(),
+		Steps: []resource.TestStep{
+			{
+				Config: materializedViewConfig(dbName, schemaName, tableName, viewName, []string{"ID", "DATA"}, fmt.Sprintf("SELECT ID, DATA FROM \\\"%s\\\";", tableName)),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "name", viewName),
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "database", dbName),
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "schema", schemaName),
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "comment", "Terraform test resource"),
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "cluster_by", "[\"ID\",\"DATA\"]"),
+					checkBool("snowflake_materialized_view.test", "is_secure", true),
+				),
+			},
+		},
+	})
+}
+
+func TestAcc_MaterializedView2(t *testing.T) {
+	dbName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	schemaName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	tableName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	viewName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+
+	resource.ParallelTest(t, resource.TestCase{
+		Providers: providers(),
+		Steps: []resource.TestStep{
+			{
+				Config: materializedViewConfig(dbName, schemaName, tableName, viewName, []string{"ID", "DATA"}, fmt.Sprintf("SELECT ID, DATA FROM \\\"%s\\\" WHERE ID LIKE 'foo%%';", tableName)),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "name", viewName),
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "database", dbName),
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "schema", schemaName),
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "comment", "Terraform test resource"),
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "cluster_by", "[\"ID\",\"DATA\"]"),
+					checkBool("snowflake_materialized_view.test", "is_secure", true),
+				),
+			},
+		},
+	})
+}
+
+func materializedViewConfig(dbName string, schemaName string, tableName string, viewName string, c []string, q string) string {
+	// convert the cluster from string slice to string
+	clusters := strings.Join(strings.Split(fmt.Sprintf("%q\n", c), " "), ",")
+	return fmt.Sprintf(`
+resource "snowflake_warehouse" "w" {
```

Comment:
You'll want to make this a required part of the materializedView resource (just like how name, database, schema are all required).